src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -475,6 +490,8 @@ private static bool IsValidConstraintType(TypeConstraintSyntax syntax, TypeWithA
                         return false;
                     }
 
+                    // "Constraint cannot be special class '{0}'"
+                    Error(diagnostics, ErrorCode.WRN_SpecialTypeAsBound, syntax, typeWithAnnotations);
```

Comment:
>WRN_SpecialTypeAsBound [](start = 49, length = 22)

Should an `object` constraint be reported as an error rather than a warning, as in C#7?

Reply:
I thought we were going to keep this as for the moment, and disallow it later to allow corefx to migrate.

---
In reply to: [289997980](https://github.com/dotnet/roslyn/pull/36109#discussion_r289997980) [](ancestors = 289997980)

Reply:
> Should an object constraint be reported as an error rather than a warning, as in C#7?

If I remember correctly, removal of support for ```object``` constraint is in the second phase. The reason is to avoid breaking CoreFx right from the first compiler upgrade.

---
In reply to: [290001495](https://github.com/dotnet/roslyn/pull/36109#discussion_r290001495) [](ancestors = 290001495,289997980)